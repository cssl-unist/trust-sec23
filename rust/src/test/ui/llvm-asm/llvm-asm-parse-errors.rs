#![feature(llvm_asm)]

fn main() {
    llvm_asm!(); //~ ERROR requires a string literal as an argument
    llvm_asm!("nop" : struct); //~ ERROR expected string literal
    llvm_asm!("mov %eax, $$0x2" : struct); //~ ERROR expected string literal
    llvm_asm!("mov %eax, $$0x2" : "={eax}" struct); //~ ERROR expected `(`
    llvm_asm!("mov %eax, $$0x2" : "={eax}"(struct)); //~ ERROR expected expression
    llvm_asm!("in %dx, %al" : "={al}"(result) : struct); //~ ERROR expected string literal
    llvm_asm!("in %dx, %al" : "={al}"(result) : "{dx}" struct); //~ ERROR expected `(`
    llvm_asm!("in %dx, %al" : "={al}"(result) : "{dx}"(struct)); //~ ERROR expected expression
    llvm_asm!("mov $$0x200, %eax" : : : struct); //~ ERROR expected string literal
    llvm_asm!("mov eax, 2" : "={eax}"(foo) : : : struct); //~ ERROR expected string literal
    llvm_asm!(123); //~ ERROR inline assembly must be a string literal
}
