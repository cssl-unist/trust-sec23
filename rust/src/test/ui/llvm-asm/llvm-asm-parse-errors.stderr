error: macro requires a string literal as an argument
  --> $DIR/llvm-asm-parse-errors.rs:4:5
   |
LL |     llvm_asm!();
   |     ^^^^^^^^^^^^ string literal required

error: expected string literal
  --> $DIR/llvm-asm-parse-errors.rs:5:23
   |
LL |     llvm_asm!("nop" : struct);
   |                       ^^^^^^ not a string literal

error: expected string literal
  --> $DIR/llvm-asm-parse-errors.rs:6:35
   |
LL |     llvm_asm!("mov %eax, $$0x2" : struct);
   |                                   ^^^^^^ not a string literal

error: expected `(`, found keyword `struct`
  --> $DIR/llvm-asm-parse-errors.rs:7:44
   |
LL |     llvm_asm!("mov %eax, $$0x2" : "={eax}" struct);
   |                                            ^^^^^^ expected `(`

error: expected expression, found keyword `struct`
  --> $DIR/llvm-asm-parse-errors.rs:8:44
   |
LL |     llvm_asm!("mov %eax, $$0x2" : "={eax}"(struct));
   |                                            ^^^^^^ expected expression

error: expected string literal
  --> $DIR/llvm-asm-parse-errors.rs:9:49
   |
LL |     llvm_asm!("in %dx, %al" : "={al}"(result) : struct);
   |                                                 ^^^^^^ not a string literal

error: expected `(`, found keyword `struct`
  --> $DIR/llvm-asm-parse-errors.rs:10:56
   |
LL |     llvm_asm!("in %dx, %al" : "={al}"(result) : "{dx}" struct);
   |                                                        ^^^^^^ expected `(`

error: expected expression, found keyword `struct`
  --> $DIR/llvm-asm-parse-errors.rs:11:56
   |
LL |     llvm_asm!("in %dx, %al" : "={al}"(result) : "{dx}"(struct));
   |                                                        ^^^^^^ expected expression

error: expected string literal
  --> $DIR/llvm-asm-parse-errors.rs:12:41
   |
LL |     llvm_asm!("mov $$0x200, %eax" : : : struct);
   |                                         ^^^^^^ not a string literal

error: expected string literal
  --> $DIR/llvm-asm-parse-errors.rs:13:50
   |
LL |     llvm_asm!("mov eax, 2" : "={eax}"(foo) : : : struct);
   |                                                  ^^^^^^ not a string literal

error: inline assembly must be a string literal
  --> $DIR/llvm-asm-parse-errors.rs:14:15
   |
LL |     llvm_asm!(123);
   |               ^^^

error: aborting due to 11 previous errors

